cli/azd/cmd/root.go

```diff
@@ -153,7 +153,10 @@ func BuildCmd[F any](
 		actionResult, err := action.Run(ctx)
 		// At this point, we know that there might be an error, so we can silence cobra from showing it after us.
 		cmd.SilenceErrors = true
-		console.MessageUxItem(ctx, actions.ToActionResult(actionResult, err))
+
+		if actionResult != nil {
+			console.MessageUxItem(ctx, actions.ToActionResult(actionResult, err))
```

Comment:
FWIW looks like the test broke because `ToActionResult` also handles printing the `err` and can be `nil`.

The other interesting part: I just noticed today that we have deviated from how `cobra` writes terminating errors to `stderr`. Our `MessageUxItem` print goes to `console.Writer` which can be `stdout` or `stderr` per command-invocation (but not both).

Reply:
Updated the logic such that we only call this if actionResult or err are non nil.  I also changed the name of `ToActionResult` to `ToUxItem` since it returns a UxItem.

> The other interesting part: I just noticed today that we have deviated from how cobra writes terminating errors to stderr. Our MessageUxItem print goes to console.Writer which can be stdout or stderr per command-invocation (but not both).

Interesting - what do you think the right long term thing here is?

I think the behavior we want is that when you are in normal terminal output mode, we write the error to stderr as a string (maybe with our color an prefix) and long term if you are using json output we write some sort of structured error object to `stdout` (because in `json` output mode we use `stderr` for progress messages).

Does that match your intuition on what we should be doing here @weikanglim and @vhvb1989?  I will make sure to open an issue to track.

Reply:
My preference would be to always have error messages in `stderr`.

When handling `json` output, the messages are typed. Thus, we can simply emit an `error` type message which is still machine readable.

We have #86 for standardization of output. We should capture it there.

Reply:
Added a note to the issue.